plugins/by-name/quarto/default.nix

```
@@ -0,0 +1,103 @@
+{ lib, pkgs, ... }:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+  inherit (lib) types;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "quarto";
+  package = "quarto-nvim";
+
+  maintainers = [ lib.maintainers.BoneyPatel ];
+
+  settingsOptions = {
+    debug = defaultNullOpts.mkBool false ''
+      Enables or disables debugging.
+    '';
+
+    closePreviewOnExit = defaultNullOpts.mkBool true ''
+      Closes preview on exit.
+    '';
+
+    lspFeatures.enabled = defaultNullOpts.mkBool true ''
+      Enables LSP features.
+    '';
+
+    lspFeatures.chunks = defaultNullOpts.mkStr "curly" ''
+      Sets chunk delimiter style.
+    '';
```

```suggestion
    lspFeatures = {
      enabled = defaultNullOpts.mkBool true ''
        Enables LSP features.
      '';

      chunks = defaultNullOpts.mkStr "curly" ''
        Sets chunk delimiter style.
      '';
      
      ...
```

Same for `codeRunner` and within `codeRunner` too.

---

Updated with nested structures for the options!